index.d.ts

```diff
@@ -140,29 +148,63 @@ export interface ContentType {
   changeFieldId (oldId: string, newId: string): void
 
   /**
-   * Changes the editor interface of given field's ID.
+   * Changes the control of given field's ID.
    *
    * @param fieldId The ID of the field.
    * @param widgetId The new widget ID for the field.
    * @param settings Widget settings
+   * @param widgetNamespace The namespace of the widget. Use 'builtin' for standard widgets or 'extension' for UI extensions.
    */
-  changeEditorInterface (fieldId: string, widgetId: string, settings?: IEditorInterfaceOptions): void
+  changeFieldControl (fieldId: string, widgetId: string, settings?: IEditorInterfaceOptions, widgetNamespace?: 'builtin' | 'extension'): void
```

Comment:
I think this is a big breaking change and we should keep the old methods as an alias to the new ones and we log a deprecation message to give the users some time to change their code and in another release we remove them entirely. WDYT?

Reply:
Yes, the actual implementation still keeps them around; how would you deprecate them?

Reply:
we can add a console.log('(deprecated) changeEditorInterface is deprecated and will be removed in a later version please use changeControl instead').
and since we are using typescript we can use something similar to [this](https://gist.github.com/michelsalib/cfff8114ce081930a453) `@deprecated('message')` tag but I am not sure if it's stable enough

Reply:
Opted to keep it simple and add a warning as well as @deprecated in the jsdoc of the index.d.ts file